docs/2.7.5/docs/query/pqs-user-guide.rst

```diff
@@ -175,52 +175,197 @@ and Operators <https://www.postgresql.org/docs/12/functions-json.html>`__ in
+
+If you need to query consistent and stable data, always use the watermark as the most recent offset in the query. If a query is just a point-in-time value that does not have global consistency requirements, you don't need the watermark.
+
+External Unsynchronized Data
+============================
+
+The previous section discussed how PQS deals with internal synchronization in its programming model. This section examines the external synchronization of data and the precautions you need to take. This section describes the time lags that the PQS client application needs to be aware of when querying data. The following figures follow the conventions described in the section `Architecture for HA and Scaling <https://docs.daml.com/deploy-daml/infrastructure-architecture/high-availability/ha-and-scaling/implementing-ha.html#architecture-for-ha-and-scaling>`__.
+
+There are four cases to consider:
+
+- The client application needs to retry on an incomplete request, such as the result of network issues.
+- PQS needs to catch up after a fresh install or restart.
+- There is a small delay for the ledger event to process.
+- A high availability PQS configuration does not have synchronized PQS instances.
+
+All distributed systems are subject to network issues. For example, networks that host the gRPC and JDBC query requests can both have intermittent issues. You must program the client application to retry for certain failures because the intermittent failure will heal itself. Most network outages are temporary, in which case the application can proceed without failing.
+
+.. image:: ./images/external-data.svg
+   :alt: A diagram showing queries using grpc and JDBC
+
+If the PQS has been (re)started or is freshly installed, then the PQS needs to catch up to the participant node's generated events. Until that happens, the PQS will not have all the data available for querying.
+
+There is a slight delay between when a participant node has finished the ledger processing and emitted an event and when the PQS finishes processing the event so that the result is available to query. For example, if the participant query performs a synchronous exercise and then immediately queries the PQS, it is remotely possible that PQS will not yet have the data. In this case, the client application needs to retry.
+
+Lastly, a highly available production deployment (see the following figure) will have multiple PQS instances, each of which processes events at slightly different rates. In this manner, the PQS instances are not synchronized, so they will process events and commit the results to the database with different times. If the client application makes a query to the PQS service which is being serviced by PQS #A and then PQS #A fails, the client application retries after a timeout, which is then serviced by PQS #B. It is possible that PQS #B will not have its watermark at the same offset as PQS #A when PQS #A failed.
```

Comment:
```suggestion
Lastly, a highly available production deployment (see the following figure) has multiple PQS instances, each of which processes events at slightly different rates. Since the PQS instances are not synchronized, they process events and commit the results to the database with different times. If the client application makes a query to the PQS service which is being serviced by PQS #A and then PQS #A fails, the client application retries after a timeout, which is then serviced by PQS #B. It is possible that PQS #B does not have its watermark at the same offset as PQS #A when PQS #A failed.
```